### PR TITLE
[#4672] Add entries to r_user_group table for "rodsadmin" and "public" groups (master)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: scripts/irods/database_upgrade.py [hunk 1]
```diff
@@ -69,6 +69,12 @@ def run_update(irods_config, cursor):
         database_connect.execute_sql_statement(cursor, "create index idx_data_main7 on R_DATA_MAIN (resc_id);")
         database_connect.execute_sql_statement(cursor, "create index idx_data_main8 on R_DATA_MAIN (data_is_dirty);")
 
+        timestamp = '{0:011d}'.format(int(time.time()))
+        rows = database_connect.execute_sql_statement(cursor, "select user_id from R_USER_MAIN where user_name in ('rodsadmin', 'public') and user_type = 'rodsgroup';").fetchall()
+        for row in rows:
+            group_id = row[0]
+            database_connect.execute_sql_statement(cursor, "insert into R_USER_GROUP values (?,?,?,?);", group_id, group_id, timestamp, timestamp)
+
```
Comment (korydraughn) [inline scripts/irods/database_upgrade.py]: Should this code be placed here?

In other words, should it be outside of `new_schema_version` if-blocks, possibly at the bottom of the file?

File: scripts/irods/database_upgrade.py [hunk 2]
```diff
@@ -69,6 +69,12 @@ def run_update(irods_config, cursor):
         database_connect.execute_sql_statement(cursor, "create index idx_data_main7 on R_DATA_MAIN (resc_id);")
         database_connect.execute_sql_statement(cursor, "create index idx_data_main8 on R_DATA_MAIN (data_is_dirty);")
```
Comment (trel) [inline scripts/irods/database_upgrade.py]: will need to bump the db version elsewhere and then check it here...
```
    elif new_schema_version == 7:
```

File: scripts/irods/database_upgrade.py [hunk 3]
```diff
@@ -69,6 +69,12 @@ def run_update(irods_config, cursor):
...
 
+        timestamp = '{0:011d}'.format(int(time.time()))
+        rows = database_connect.execute_sql_statement(cursor, "select user_id from R_USER_MAIN where user_name in ('rodsadmin', 'public') and user_type_name = 'rodsgroup';").fetchall()
```
Comment (trel) [inline scripts/irods/database_upgrade.py]: don't we want a query to find accounts that are *NOT* in the `R_USER_GROUP` table?  so, if they had already 'fixed' this manually, there would be nothing to do.  and if they had somehow added others... they would get found here as well?

Comment (trel) [inline scripts/irods/database_upgrade.py]: Codacy Issue found: [line too long (184 > 159 characters)](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893643)

File: scripts/irods/database_upgrade.py [hunk 4]
```diff
@@ -69,6 +69,12 @@ def run_update(irods_config, cursor):
         database_connect.execute_sql_statement(cursor, "create index idx_data_main7 on R_DATA_MAIN (resc_id);")
         database_connect.execute_sql_statement(cursor, "create index idx_data_main8 on R_DATA_MAIN (data_is_dirty);")
 
+        timestamp = '{0:011d}'.format(int(time.time()))
```
Comment (trel) [inline scripts/irods/database_upgrade.py]: Codacy Issue found: [undefined name 'time'](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4893643)

File: scripts/irods/database_upgrade.py (hunk 3 above)
Comment (korydraughn) [reply scripts/irods/database_upgrade.py]: Will adjust it.

File: scripts/irods/database_upgrade.py (hunk 2 above)
Comment (korydraughn) [reply scripts/irods/database_upgrade.py]: Is the bumping of the db version supposed to be part of this PR? Not sure about that, but if-else has been added.

File: scripts/irods/database_upgrade.py (hunk 3 above)
Comment (korydraughn) [reply scripts/irods/database_upgrade.py]: the SQL has been adjusted and should handle all cases gracefully.

File: scripts/irods/database_upgrade.py (hunk 2 above)
Comment (trel) [reply scripts/irods/database_upgrade.py]: yes, i think so - if we're bumping the logic to 7... we need to tell 4.2.8 the db should be at 7.    you could double tag a separate commit with this issue and the 4.2.8 release activities - so it's easy to figure out later.

**Comment by korydraughn** [comment]
Upgrade from vanilla 4.2.2 to 4.3.0 worked. The missing rows were added to the table.

**Comment by korydraughn** [comment]
Upgrade from 4.2.2 with all rows fixed to 4.3.0 worked as well. The upgrade script did not attempt to add any additional rows.

**Comment by korydraughn** [comment]
Fresh install of 4.3.0 also worked (no previously installed version).

I think this is ready.

**Comment by trel** [comment]
awesome - let's # it.

**Pull request merged**